lib/src/clojure_lsp/db.clj

```diff
@@ -104,3 +106,98 @@
   (-> (read-global-cache)
       (db-change-fn)
       (upsert-global-cache!)))
+
+;; DATA
+
+(defn entry->normalized-entries [{:keys [bucket] :as element}]
+  (cond
+    ;; We create two entries here (and maybe more for refer)
+    (identical? :namespace-usages bucket)
+    (cond-> [(set/rename-keys element {:to :name})]
+      (:alias element)
+      (conj (set/rename-keys (assoc element :bucket :namespace-alias) {:alias-row :name-row
+                                                                       :alias-col :name-col
+                                                                       :alias-end-row :name-end-row
+                                                                       :alias-end-col :name-end-col})))
+
+    (contains? #{:locals :local-usages :keywords} bucket)
+    [(-> element
+         (assoc :name-row (or (:name-row element) (:row element))
+                :name-col (or (:name-col element) (:col element))
+                :name-end-row (or (:name-end-row element) (:end-row element))
+                :name-end-col (or (:name-end-col element) (:end-col element))))]
+
+    (identical? :java-class-definitions bucket)
+    [(-> element
+         (dissoc :uri)
+         (assoc :name-row 0
+                :name-col 0
+                :name-end-row 0
+                :name-end-col 0))]
+
+    (identical? :java-class-usages bucket)
+    [(-> element
+         (dissoc :uri)
+         (assoc :name-row (or (:name-row element) (:row element))
+                :name-col (or (:name-col element) (:col element))
+                :name-end-row (or (:name-end-row element) (:end-row element))
+                :name-end-col (or (:name-end-col element) (:end-col element))))]
+
+    :else
+    [element]))
+
+(defn ^:private valid-element? [{:keys [name-row name-col name-end-row name-end-col]}]
+  (and name-row
+       name-col
+       name-end-row
+       name-end-col))
+
+(defn ^:private normalize-analysis [external? analysis]
+  (for [[bucket vs] analysis
+        v vs
+        element (entry->normalized-entries (assoc v :bucket bucket :external? external?))
+        :when (valid-element? element)]
+    element))
+
+(defn normalize-kondo
```

Comment:
I was trying to avoid couple the db namespace with kondo, maybe we can move those specific kondo functions to `kondo.clj`?

Reply:
@ericdallo, now all the `clojure-lsp.kondo/run-*!` functions return normalized data.

Generally, I think this is cleaner, but has one small drawback in terms of the plans to cache project analysis. In the first patch, the benefit of coupling the db ns to kondo was that `:external?` was being passed into `db/with-kondo-results`. The db ns could have used that key to decide whether it should cache the new analysis. Now it'll be a bit more work to make that decision. It's an easy problem to fix, so I'm not worried about it, and generally I'm happier with how responsibilities are divided up at the moment.

One final question—the db namespace still has a function called `merge-kondo-results`. Would you prefer if it doesn't use the word "kondo" at all?

Reply:
not sure, there is a :kondo-config key [here](https://github.com/clojure-lsp/clojure-lsp/pull/938) as well, so not sure, still reviewing the code, but I'd like too keep all/most of kondo specific things on kondo.clj, although we are quite coupled to kondo analysis structure on multiple places already 🤷🏻‍♂️

Reply:
Moved `db/merge-kondo-results` to `kondo/db-with-results`.